src/org/javarosa/model/xform/SMSSerializingVisitor.java

```diff
@@ -133,83 +127,83 @@ public IDataPayload createSerializedPayload(FormInstance model, IDataReference r
     public void visit(FormInstance tree) {
         nodeSet = new String();
 
-        //TreeElement root = tree.getRoot();
-        TreeElement root = tree.resolveReference(rootRef);
+        TreeElement root = tree.getRoot();
```

Comment:
Just checking that you really meant to uncomment this line and remove the line below it. If so, what does that change do?

Reply:
I pulled in those changes from the Medic branch https://github.com/medic/javarosa/commit/90c46a6a3fc19b1b75783cb27cfa907d37099cca#diff-700d0f975ca498a121d65ff76e95a8c7
So it's probably can be reversed once it's not something that has to be done.

Reply:
I should probably read more of the background to see how we got to where we are.